### PR TITLE
[14.0][ADD] stock_request_partner: new module to allow to select partner in stock request

### PR DESCRIPTION
**Pull request by alan196**
### Proposed changes

I've added the module stock_request_partner, this module allows to define a partner in a stock request and stock request order.

### Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Migration Update

### Checklist

- [x] I have read the CONTRIBUTING doc
- [x] I have signed the CLA
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation

### Comments

This PR depends on https://github.com/OCA/stock-logistics-warehouse/pull/985

### CODE REVIEW

**Review by olveracuenca — APPROVED** [review]
LGTM

**Comment by rousseldenis** [comment]
@alan196 Could you rebase ?

**Comment by rousseldenis** [comment]
@alan196

**Comment by alan196** [comment]
@rousseldenis rebased 👍 sorry for the late response

File: stock_request_partner/readme/USAGE.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## Creation
+
+* Go to 'Stock Requests / Stock Requests' and create a new Request.
+* Indicate a partner, product, quantity and location.
+* Press 'Confirm'.
+
+In case that transfers are created, the partner will defined in the related transfers.
```
Comment (rousseldenis) [inline stock_request_partner/readme/USAGE.rst]: @alan196  You maybe want to speak about procurement group auto creation if a partner is set ?

Comment (alan196) [reply stock_request_partner/readme/USAGE.rst]: @rousseldenis 👍 readme updated

**Review by rousseldenis — APPROVED** [review]
LGTM

**Pull request merged**